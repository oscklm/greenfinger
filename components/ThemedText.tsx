import { Text, type TextProps } from "react-native";

import {
  createStyleSheet,
  UnistylesVariants,
  useStyles,
} from "react-native-unistyles";

type ThemeTextVariants = UnistylesVariants<typeof stylesheet>;

export type ThemedTextProps = TextProps & ThemeTextVariants;

export function ThemedText({ variant, ...rest }: ThemedTextProps) {
  const { styles } = useStyles(stylesheet, {
    variant,
  });

  return <Text style={styles.default} {...rest} />;
}

const stylesheet = createStyleSheet((theme, rt) => ({
  default: {
    fontSize: 16,
    lineHeight: 24,
    variants: {
      variant: {
        defaultSemiBold: {
          fontSize: 16,
          lineHeight: 24,
          fontWeight: "600",
        },
        title: {
          fontSize: 32,
          fontWeight: "bold",
          lineHeight: 32,
        },
        subtitle: {
          fontSize: 20,
          fontWeight: "bold",
        },
        link: {
          lineHeight: 30,
          fontSize: 16,
          color: "#0a7ea4",
        },
      },
    },
  },
}));
