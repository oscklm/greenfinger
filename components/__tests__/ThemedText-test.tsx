import * as React from 'react';
import { render } from '@testing-library/react-native';
import { ThemedText } from '../ThemedText';

it(`renders correctly`, () => {
  const { toJSON, getByText } = render(<ThemedText>Snapshot test!</ThemedText>);
  
  expect(getByText('Snapshot test!')).toBeTruthy(); // Ensuring the text is rendered
  expect(toJSON()).toMatchSnapshot();
});
