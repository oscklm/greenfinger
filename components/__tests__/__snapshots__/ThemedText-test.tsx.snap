// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    {
      "fontSize": 16,
      "lineHeight": 24,
    }
  }
>
  Snapshot test!
</Text>
`;
